`pkgsStatic.lixPackageSets.git.lix` fails with errors like:
```
FAILED: lix/nix/nix 
x86_64-unknown-linux-musl-clang++  -o lix/nix/nix lix/nix/nix.p/add-to-store.cc.o lix/nix/nix.p/app.cc.o lix/nix/nix.p/build.cc.o lix/nix/nix.p/bundle.cc.o lix/nix/nix.p/cat.cc.o lix/nix/nix.p/config.cc.o lix/nix/nix.p/copy.cc.o lix/nix/nix.p/daemon.cc.o lix/nix/nix.p/derivation-add.cc.o lix/nix/nix.p/derivation-show.cc.o lix/nix/nix.p/derivation.cc.o lix/nix/nix.p/develop.cc.o lix/nix/nix.p/diff-closures.cc.o lix/nix/nix.p/doctor.cc.o lix/nix/nix.p/dump-path.cc.o lix/nix/nix.p/edit.cc.o lix/nix/nix.p/eval.cc.o lix/nix/nix.p/flake.cc.o lix/nix/nix.p/fmt.cc.o lix/nix/nix.p/hash.cc.o lix/nix/nix.p/log.cc.o lix/nix/nix.p/ls.cc.o lix/nix/nix.p/main.cc.o lix/nix/nix.p/make-content-addressed.cc.o lix/nix/nix.p/nar.cc.o lix/nix/nix.p/optimise-store.cc.o lix/nix/nix.p/path-from-hash-part.cc.o lix/nix/nix.p/path-info.cc.o lix/nix/nix.p/ping-store.cc.o lix/nix/nix.p/prefetch.cc.o lix/nix/nix.p/profile.cc.o lix/nix/nix.p/realisation.cc.o lix/nix/nix.p/registry.cc.o lix/nix/nix.p/repl.cc.o lix/nix/nix.p/run.cc.o lix/nix/nix.p/search.cc.o lix/nix/nix.p/sigs.cc.o lix/nix/nix.p/store-copy-log.cc.o lix/nix/nix.p/store-delete.cc.o lix/nix/nix.p/store-gc.cc.o lix/nix/nix.p/store-repair.cc.o lix/nix/nix.p/store.cc.o lix/nix/nix.p/upgrade-nix.cc.o lix/nix/nix.p/verify.cc.o lix/nix/nix.p/why-depends.cc.o lix/nix/nix.p/.._legacy_build-remote.cc.o lix/nix/nix.p/.._legacy_dotgraph.cc.o lix/nix/nix.p/.._legacy_graphml.cc.o lix/nix/nix.p/.._legacy_nix-build.cc.o lix/nix/nix.p/.._legacy_nix-channel.cc.o lix/nix/nix.p/.._legacy_nix-collect-garbage.cc.o lix/nix/nix.p/.._legacy_nix-copy-closure.cc.o lix/nix/nix.p/.._legacy_nix-env.cc.o lix/nix/nix.p/.._legacy_nix-instantiate.cc.o lix/nix/nix.p/.._legacy_nix-store.cc.o lix/nix/nix.p/.._legacy_user-env.cc.o -Wl,--as-needed -Wl,--no-undefined -Wl,--whole-archive -Wl,--start-group lix/liblibasanoptions.a -Wl,--no-whole-archive -fsanitize=signed-integer-overflow -fsanitize-undefined-trap-on-error -shared-libsan -pthread -Wl,--no-copy-dt-needed-entries -Wl,-rpath,XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX lix/libutil/liblixutil.a lix/libstore/liblixstore.a lix/libfetchers/liblixfetchers.a lix/libexpr/liblixexpr.a lix/libmain/liblixmain.a lix/libcmd/liblixcmd.a lix/lix-doc/liblix_doc.a /nix/store/z8lh6xj0zlriby46ask43r8ad5w112xs-capnproto-static-x86_64-unknown-linux-musl-1.2.0/lib/libkj-async.a /nix/store/z8lh6xj0zlriby46ask43r8ad5w112xs-capnproto-static-x86_64-unknown-linux-musl-1.2.0/lib/libkj.a /nix/store/b5r4hw3gsb9db785l867mr8ckflkchjg-libarchive-static-x86_64-unknown-linux-musl-3.8.1-lib/lib/libarchive.a /nix/store/sp6d6xjc541vd0dvyi7h4wy1fx1yy2q1-openssl-static-x86_64-unknown-linux-musl-3.5.1/lib/libcrypto.a /nix/store/x4hsb41rjhfxsl2s22mdni3g69ig17w2-xz-static-x86_64-unknown-linux-musl-5.8.1/lib/liblzma.a -lpthread /nix/store/54qzw5h35186a8bb02fb3x8nalszhpi5-libxml2-static-x86_64-unknown-linux-musl-2.14.4-unstable-2025-06-20/lib/libxml2.a -lm -ldl -L/nix/store/sp6d6xjc541vd0dvyi7h4wy1fx1yy2q1-openssl-static-x86_64-unknown-linux-musl-3.5.1/lib -L/nix/store/b5r4hw3gsb9db785l867mr8ckflkchjg-libarchive-static-x86_64-unknown-linux-musl-3.8.1-lib/lib -L/nix/store/x4hsb41rjhfxsl2s22mdni3g69ig17w2-xz-static-x86_64-unknown-linux-musl-5.8.1/lib -L/nix/store/54qzw5h35186a8bb02fb3x8nalszhpi5-libxml2-static-x86_64-unknown-linux-musl-2.14.4-unstable-2025-06-20/lib -lacl -lzstd -lbz2 -lz /nix/store/xalxwybk1gn1gmi7mi45qrdvpkidndb0-boehm-gc-static-x86_64-unknown-linux-musl-8.2.8/lib/libgc.a /nix/store/z8lh6xj0zlriby46ask43r8ad5w112xs-capnproto-static-x86_64-unknown-linux-musl-1.2.0/lib/libcapnp-rpc.a /nix/store/z8lh6xj0zlriby46ask43r8ad5w112xs-capnproto-static-x86_64-unknown-linux-musl-1.2.0/lib/libcapnp.a /nix/store/623b15b6p3qi3x0b7lrmz3invd57qix4-brotli-static-x86_64-unknown-linux-musl-1.1.0-lib/lib/libbrotlicommon.a /nix/store/623b15b6p3qi3x0b7lrmz3invd57qix4-brotli-static-x86_64-unknown-linux-musl-1.1.0-lib/lib/libbrotlidec.a /nix/store/623b15b6p3qi3x0b7lrmz3invd57qix4-brotli-static-x86_64-unknown-linux-musl-1.1.0-lib/lib/libbrotlienc.a /nix/store/w9024j0i1l2p1wp0zlxzmcn9zcs203ic-libcpuid-static-x86_64-unknown-linux-musl-0.8.0/lib/libcpuid.a -L/nix/store/sp6d6xjc541vd0dvyi7h4wy1fx1yy2q1-openssl-static-x86_64-unknown-linux-musl-3.5.1/lib -L/nix/store/b5r4hw3gsb9db785l867mr8ckflkchjg-libarchive-static-x86_64-unknown-linux-musl-3.8.1-lib/lib -L/nix/store/x4hsb41rjhfxsl2s22mdni3g69ig17w2-xz-static-x86_64-unknown-linux-musl-5.8.1/lib -L/nix/store/54qzw5h35186a8bb02fb3x8nalszhpi5-libxml2-static-x86_64-unknown-linux-musl-2.14.4-unstable-2025-06-20/lib -lacl -lzstd -lbz2 -lz -L/nix/store/sp6d6xjc541vd0dvyi7h4wy1fx1yy2q1-openssl-static-x86_64-unknown-linux-musl-3.5.1/lib -L/nix/store/b5r4hw3gsb9db785l867mr8ckflkchjg-libarchive-static-x86_64-unknown-linux-musl-3.8.1-lib/lib -L/nix/store/x4hsb41rjhfxsl2s22mdni3g69ig17w2-xz-static-x86_64-unknown-linux-musl-5.8.1/lib -L/nix/store/54qzw5h35186a8bb02fb3x8nalszhpi5-libxml2-static-x86_64-unknown-linux-musl-2.14.4-unstable-2025-06-20/lib -lacl -lzstd -lbz2 -lz /nix/store/qiamim7zrjv4dj1pwx4gmpyr52xsf2l7-curl-static-x86_64-unknown-linux-musl-8.14.1/lib/libcurl.a /nix/store/xs5z9i17ns2na47kfyv3x7la3kqyachz-nghttp2-static-x86_64-unknown-linux-musl-1.65.0-lib/lib/libnghttp2.a /nix/store/a83k8yhkk5r77bw7590j9vc1lg06l1w8-libidn2-static-x86_64-unknown-linux-musl-2.3.8/lib/libidn2.a /nix/store/mvdkgbwcac1k0p1zqqaq6v3apm4sp2bq-libssh2-static-x86_64-unknown-linux-musl-1.11.1/lib/libssh2.a /nix/store/17xy7m7lh883jx5xpm1ld0zhgp8w71c6-libpsl-static-x86_64-unknown-linux-musl-0.21.5/lib/libpsl.a /nix/store/sp6d6xjc541vd0dvyi7h4wy1fx1yy2q1-openssl-static-x86_64-unknown-linux-musl-3.5.1/lib/libssl.a /nix/store/40kja02wfnw1gdlrcyh75c7x7iibrwvn-zstd-static-x86_64-unknown-linux-musl-1.5.7/lib/libzstd.a /nix/store/8r4xf3v70mqdl8f3jy1y5zjgwrd4x4ym-zlib-static-x86_64-unknown-linux-musl-1.3.1/lib/libz.a -L/nix/store/8r4xf3v70mqdl8f3jy1y5zjgwrd4x4ym-zlib-static-x86_64-unknown-linux-musl-1.3.1/lib -L/nix/store/40kja02wfnw1gdlrcyh75c7x7iibrwvn-zstd-static-x86_64-unknown-linux-musl-1.5.7/lib -L/nix/store/qiamim7zrjv4dj1pwx4gmpyr52xsf2l7-curl-static-x86_64-unknown-linux-musl-8.14.1/lib -L/nix/store/sp6d6xjc541vd0dvyi7h4wy1fx1yy2q1-openssl-static-x86_64-unknown-linux-musl-3.5.1/lib -L/nix/store/17xy7m7lh883jx5xpm1ld0zhgp8w71c6-libpsl-static-x86_64-unknown-linux-musl-0.21.5/lib -L/nix/store/a83k8yhkk5r77bw7590j9vc1lg06l1w8-libidn2-static-x86_64-unknown-linux-musl-2.3.8/lib -L/nix/store/xs5z9i17ns2na47kfyv3x7la3kqyachz-nghttp2-static-x86_64-unknown-linux-musl-1.65.0-lib/lib -L/nix/store/mvdkgbwcac1k0p1zqqaq6v3apm4sp2bq-libssh2-static-x86_64-unknown-linux-musl-1.11.1/lib -L/nix/store/rr598iz9x2p5ll8l97mm7pkgdbqmcsl9-libssh2-static-x86_64-unknown-linux-musl-1.11.1-dev/lib -lunistring /nix/store/g12lm22lmymf4rdj5v1h9irwr95pqk77-libseccomp-static-x86_64-unknown-linux-musl-2.6.0-lib/lib/libseccomp.a /nix/store/z9n2156vfm07nycrcd7z04sjgdypijkf-libsodium-static-x86_64-unknown-linux-musl-1.0.20/lib/libsodium.a /nix/store/3ckxla4d8hsx7mb4mdn5y0chg8d7qzy3-sqlite-static-x86_64-unknown-linux-musl-3.50.2/lib/libsqlite3.a -L/nix/store/3ckxla4d8hsx7mb4mdn5y0chg8d7qzy3-sqlite-static-x86_64-unknown-linux-musl-3.50.2/lib -lz -L/nix/store/sp6d6xjc541vd0dvyi7h4wy1fx1yy2q1-openssl-static-x86_64-unknown-linux-musl-3.5.1/lib -L/nix/store/b5r4hw3gsb9db785l867mr8ckflkchjg-libarchive-static-x86_64-unknown-linux-musl-3.8.1-lib/lib -L/nix/store/x4hsb41rjhfxsl2s22mdni3g69ig17w2-xz-static-x86_64-unknown-linux-musl-5.8.1/lib -L/nix/store/54qzw5h35186a8bb02fb3x8nalszhpi5-libxml2-static-x86_64-unknown-linux-musl-2.14.4-unstable-2025-06-20/lib -lacl -lzstd -lbz2 -lz -L/nix/store/sp6d6xjc541vd0dvyi7h4wy1fx1yy2q1-openssl-static-x86_64-unknown-linux-musl-3.5.1/lib -L/nix/store/b5r4hw3gsb9db785l867mr8ckflkchjg-libarchive-static-x86_64-unknown-linux-musl-3.8.1-lib/lib -L/nix/store/x4hsb41rjhfxsl2s22mdni3g69ig17w2-xz-static-x86_64-unknown-linux-musl-5.8.1/lib -L/nix/store/54qzw5h35186a8bb02fb3x8nalszhpi5-libxml2-static-x86_64-unknown-linux-musl-2.14.4-unstable-2025-06-20/lib -lacl -lzstd -lbz2 -lz -L/nix/store/sp6d6xjc541vd0dvyi7h4wy1fx1yy2q1-openssl-static-x86_64-unknown-linux-musl-3.5.1/lib -L/nix/store/b5r4hw3gsb9db785l867mr8ckflkchjg-libarchive-static-x86_64-unknown-linux-musl-3.8.1-lib/lib -L/nix/store/x4hsb41rjhfxsl2s22mdni3g69ig17w2-xz-static-x86_64-unknown-linux-musl-5.8.1/lib -L/nix/store/54qzw5h35186a8bb02fb3x8nalszhpi5-libxml2-static-x86_64-unknown-linux-musl-2.14.4-unstable-2025-06-20/lib -lacl -lzstd -lbz2 -lz -L/nix/store/sp6d6xjc541vd0dvyi7h4wy1fx1yy2q1-openssl-static-x86_64-unknown-linux-musl-3.5.1/lib -L/nix/store/b5r4hw3gsb9db785l867mr8ckflkchjg-libarchive-static-x86_64-unknown-linux-musl-3.8.1-lib/lib -L/nix/store/x4hsb41rjhfxsl2s22mdni3g69ig17w2-xz-static-x86_64-unknown-linux-musl-5.8.1/lib -L/nix/store/54qzw5h35186a8bb02fb3x8nalszhpi5-libxml2-static-x86_64-unknown-linux-musl-2.14.4-unstable-2025-06-20/lib -lacl -lzstd -lbz2 -lz /nix/store/xvwkab4bym338r29hqabhnh45ab6lp9c-editline-static-x86_64-unknown-linux-musl-1.17.1/lib/libeditline.a /nix/store/i571p6fapdkp2mlarjbkfqcpgbcjr4fq-lowdown-static-x86_64-unknown-linux-musl-2.0.2-lib/lib/liblowdown.a /nix/store/q1qsj9zr3r5hxpp2nb1ks2zblp8ixckc-ncurses-static-x86_64-unknown-linux-musl-6.5/lib/libncursesw.a -lunwind -Wl,--end-group
clang++: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
/nix/store/0adymgz38j1xgca61fgg14lvaq0iwq1b-x86_64-unknown-linux-musl-binutils-2.44/bin/x86_64-unknown-linux-musl-ld: /nix/store/z8lh6xj0zlriby46ask43r8ad5w112xs-capnproto-static-x86_64-unknown-linux-musl-1.2.0/lib/libkj-async.a(async.c++.o): in function `kj::_::FiberStack::FiberStack(unsigned long)':
(.text+0x45b8): undefined reference to `makecontext'
```

Which is caused by capnproto shipping broken pkg-config files.

--- a/pkgs/by-name/ca/capnproto/fix-libucontext.patch
+++ b/pkgs/by-name/ca/capnproto/fix-libucontext.patch
@@ -1,0 +1,74 @@
+From f26dd335c8650a2f8ab7d6e4fb5dfc40ee6af618 Mon Sep 17 00:00:00 2001
+From: Jade Lovelace <software@lfcode.ca>
+Date: Sun, 10 Aug 2025 10:54:14 +0000
+Subject: [PATCH] fix: include libucontext in Requires.private in cmake builds
+
+This is required so that static musl actually links to libucontext
+correctly to get setcontext/etc for fibers.
+---
+ c++/CMakeLists.txt           | 4 ++++
+ c++/configure.ac             | 4 ++++
+ c++/pkgconfig/kj-async.pc.in | 1 +
+ 3 files changed, 9 insertions(+)
+
+diff --git a/c++/CMakeLists.txt b/c++/CMakeLists.txt
+index f335f12f7c..b2a24e40e0 100644
+--- a/c++/CMakeLists.txt
++++ b/c++/CMakeLists.txt
+@@ -96,6 +96,9 @@ set_property(CACHE WITH_FIBERS PROPERTY STRINGS AUTO ON OFF)
+ # CapnProtoConfig.cmake.in needs this variable.
+ set(_WITH_LIBUCONTEXT OFF)
+ 
++# Used by pkg-config files
++set(ASYNC_REQUIRES_PRIVATE "")
++
+ if (WITH_FIBERS OR WITH_FIBERS STREQUAL "AUTO")
+   set(_capnp_fibers_found OFF)
+   if (WIN32 OR CYGWIN)
+@@ -116,6 +119,7 @@ if (WITH_FIBERS OR WITH_FIBERS STREQUAL "AUTO")
+         if (libucontext_FOUND)
+           set(_WITH_LIBUCONTEXT ON)
+           set(_capnp_fibers_found ON)
++          set(ASYNC_REQUIRES_PRIVATE "${ASYNC_REQUIRES_PRIVATE} libucontext")
+         endif()
+       else()
+         set(_capnp_fibers_found OFF)
+diff --git a/c++/configure.ac b/c++/configure.ac
+index a2de7aac80..ce3c632e8c 100644
+--- a/c++/configure.ac
++++ b/c++/configure.ac
+@@ -216,6 +216,8 @@ AS_IF([test "$with_fibers" != no], [
+   ])
+ ])
+ 
++ASYNC_REQUIRES_PRIVATE=""
++
+ # Check for library support necessary for fibers.
+ AS_IF([test "$with_fibers" != no], [
+   case "${host_os}" in
+@@ -241,6 +243,7 @@ AS_IF([test "$with_fibers" != no], [
+         ])
+         AS_IF([test "$ucontext_supports_fibers" = yes], [
+           ASYNC_LIBS="$ASYNC_LIBS -lucontext"
++          ASYNC_REQUIRES_PRIVATE="$ASYNC_REQUIRES_PRIVATE libucontext"
+           with_fibers=yes
+         ], [
+           AS_IF([test "$with_fibers" = yes], [
+@@ -259,6 +262,7 @@ AS_IF([test "$with_fibers" = yes], [
+ ], [
+   CXXFLAGS="$CXXFLAGS -DKJ_USE_FIBERS=0"
+ ])
++AC_SUBST(ASYNC_REQUIRES_PRIVATE, $ASYNC_REQUIRES_PRIVATE)
+ 
+ # CapnProtoConfig.cmake.in needs these variables,
+ # we force them to NO because we don't need the CMake dependency for them,
+diff --git a/c++/pkgconfig/kj-async.pc.in b/c++/pkgconfig/kj-async.pc.in
+index 49d5ff6996..41aae28555 100644
+--- a/c++/pkgconfig/kj-async.pc.in
++++ b/c++/pkgconfig/kj-async.pc.in
+@@ -8,4 +8,5 @@ Description: Basic utility library called KJ (async part)
+ Version: @VERSION@
+ Libs: -L${libdir} -lkj-async @ASYNC_LIBS@ @PTHREAD_CFLAGS@ @PTHREAD_LIBS@ @STDLIB_FLAG@
+ Requires: kj = @VERSION@
++Requires.private: @ASYNC_REQUIRES_PRIVATE@
+ Cflags: -I${includedir} @ASYNC_LIBS@ @PTHREAD_CFLAGS@ @STDLIB_FLAG@ @CAPNP_LITE_FLAG@

--- a/pkgs/by-name/ca/capnproto/package.nix
+++ b/pkgs/by-name/ca/capnproto/package.nix
@@ -1,11 +1,14 @@
 {
   binutils,
   lib,
+  libucontext,
+  pkg-config,
   clangStdenv,
   fetchFromGitHub,
   cmake,
   openssl,
   zlib,
+  nix-update-script,
 }:
 
 let
@@ -40,12 +43,22 @@ clangStdenv.mkDerivation rec {
     hash = "sha256-aDcn4bLZGq8915/NPPQsN5Jv8FRWd8cAspkG3078psc=";
   };
 
-  nativeBuildInputs = [ cmake ];
+  patches = [
+    # https://github.com/capnproto/capnproto/pull/2377
+    ./fix-libucontext.patch
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
   propagatedBuildInputs = [
     openssl
     zlib
   ]
-  ++ lib.optional (clangStdenv.cc.isClang && clangStdenv.hostPlatform.isStatic) empty-libgcc_eh;
+  ++ lib.optional (clangStdenv.cc.isClang && clangStdenv.hostPlatform.isStatic) empty-libgcc_eh
+  # musl doesn't ship getcontext/setcontext unlike basically every other libc
+  ++ lib.optional clangStdenv.hostPlatform.isMusl libucontext;
 
   # FIXME: separate the binaries from the stuff that user systems actually use
   # This runs into a terrible UX issue in Lix and I just don't want to debug it
@@ -55,6 +68,10 @@ clangStdenv.mkDerivation rec {
 
   cmakeFlags = [
     (lib.cmakeBool "BUILD_SHARED_LIBS" true)
+    # merely requires setcontext/getcontext (libc), lix expects fibers to
+    # be available, and we want to make sure that the build will fail if
+    # it breaks
+    (lib.cmakeBool "WITH_FIBERS" true)
     # Take optimization flags from CXXFLAGS rather than cmake injecting them
     (lib.cmakeFeature "CMAKE_BUILD_TYPE" "None")
   ];
@@ -65,6 +82,8 @@ clangStdenv.mkDerivation rec {
   };
 
   separateDebugInfo = true;
+
+  passthru.updateScript = nix-update-script { };
 
   meta = with lib; {
     homepage = "https://capnproto.org/";

